src/witan/cic/tuairisc/charting.clj

```diff
@@ -25,93 +73,130 @@
+                                    :type  "quantitative"
+                                    :axis  {:format y-format}
+                                    :scale {:domain y-scale :zero y-zero}}}
+                 :layer    [{:mark {:type  "line"
+                                    :size  5
+                                    :point {:filled      true #_false
+                                            ;; :size        150
+                                            :strokewidth 0.5}}}
+                            {:mark     "errorband"
+                             :encoding {:y       {:field oru :title y-title :type "quantitative"}
+                                        :y2      {:field orl}
+                                        :color   {:field group :title group-title}
+                                        :tooltip nil}}
+                            {:mark     "errorband"
+                             :encoding {:y       {:field iru :title y-title :type "quantitative"}
+                                        :y2      {:field irl}
+                                        :color   {:field group :title group-title}
+                                        :tooltip nil}}
+                            {:transform [{:filter {:param "hover" :empty false}}] :mark {:type "point" :size 200 :strokeWidth 5}}]}
+                {:data     {:values (-> data
+                                        tooltip-formatf
+                                        (tc/order-by [x])
+                                        (tc/select-columns [x group tooltip-field])
+                                        (tc/map-columns group [group] tooltip-group-formatf)
+                                        (tc/pivot->wider [group] [tooltip-field] {:drop-missing? false})
+                                        (tc/replace-missing :all :value "")
+                                        (tc/reorder-columns (cons x (:domain-value colors-and-shapes)))
+                                        (tc/rows :as-maps))}
+                 :mark     {:type "rule" :strokeWidth 4}
+                 :encoding {:opacity {:condition {:value 0.3 :param "hover" :empty false}
+                                      :value     0}
+                            :tooltip (into [{:field x :type "temporal" :format x-format :title x-title}]
+                                           (map (fn [g] {:field (tooltip-group-formatf g)}))
+                                           (keep (into #{} (get data group)) (get colors-and-shapes :domain-value)))}
+                 :params   [{:name   "hover"
+                             :select {:type    "point"
+                                      ;; :size    200
+                                      :fields  [x]
+                                      :nearest true
+                                      :on      "pointerover"
+                                      :clear   "pointerout"}}]}]}))
 
 (defn line-and-ribbon-plot
```

Comment:
Why not updating `line-and-ribbon-plot` & `line-shape-and-ribbon-plot` to also match `send.adroddiad.vega-specs.lines`?